vllm/model_executor/models/minicpmv.py

```diff
@@ -430,248 +501,101 @@ def __init__(
-            from transformers.models.idefics2.modeling_idefics2 import (
-                Idefics2VisionTransformer)
-            model = Idefics2VisionTransformer(self.config.vision_config)
-            if self.config.drop_vision_last_layer:
-                model.encoder.layers = model.encoder.layers[:-1]
-        else:
-            from vllm.model_executor.models.na_vit import (
-                SiglipVisionTransformer)
-            if self.config._attn_implementation == 'flash_attention_2':
-                self.config.vision_config._attn_implementation \
-                    = 'flash_attention_2'
-            else:
-                # not support sdpa
-                self.config.vision_config._attn_implementation = 'eager'
-            model = SiglipVisionTransformer(self.config.vision_config)
-            if self.config.drop_vision_last_layer:
-                model.encoder.layers = model.encoder.layers[:-1]
-        return model
-
-    def init_resampler(self, embed_dim: int, vision_dim: int):
-        default_dtype = torch.get_default_dtype()
-        torch.set_default_dtype(torch.float16)
-        if self.version == (2, 0):
-            resampler = Resampler(grid_size=int(
-                math.sqrt(self.config.query_num)),
-                                  num_queries=None,
-                                  embed_dim=embed_dim,
-                                  num_heads=embed_dim // 128,
-                                  kv_dim=vision_dim,
-                                  adaptive=True,
-                                  version=self.version)
+    def get_embedding(
+        self,
+        input_ids: torch.Tensor,
+        image_inputs: Optional[MiniCPMVImageInputs],
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        vlm_embedding: torch.Tensor = self.llm.embed_tokens(input_ids)
+        if hasattr(self.config, "scale_emb"):
+            vlm_embedding *= self.config.scale_emb
+
+        if image_inputs is None:  # No image
+            vision_hidden_states = torch.tensor([], device=input_ids.device)
         else:
-            resampler = Resampler(num_queries=self.config.query_num,
-                                  grid_size=None,
-                                  embed_dim=embed_dim,
-                                  num_heads=embed_dim // 128,
-                                  kv_dim=vision_dim,
-                                  adaptive=True,
-                                  version=self.version)
-        torch.set_default_dtype(default_dtype)
-        return resampler
+            vision_hidden_states = self.get_vision_hidden_states(image_inputs)
+
+            # See NOTE in _parse_and_validate_inputs
+            if len(image_inputs["image_bounds"]) > 0:
+                image_indices = torch.stack([
+                    torch.arange(r[0], r[1], dtype=torch.long)
+                    for r in image_inputs["image_bounds"]
+                ]).to(vlm_embedding.device, vlm_embedding.dtype)
```

Comment:
Just put the `dtype` of `image_indices` to `torch.int64`. It is not supposed to be `torch.float`

Reply:
Fixed.